src/xcode/ENA/ENA/Source/AppDelegate.swift

```diff
@@ -18,20 +19,296 @@ protocol CoronaWarnAppDelegate: AnyObject {
 	var taskScheduler: ENATaskScheduler { get }
 	var serverEnvironment: ServerEnvironment { get }
 	var contactDiaryStore: ContactDiaryStoreV1 { get }
-}
 
-extension AppDelegate: CoronaWarnAppDelegate {
-	// required - otherwise app will crash because cast will fail.
+	func requestUpdatedExposureState()
+
 }
 
-extension AppDelegate {
+@UIApplicationMain
+// swiftlint:disable:next type_body_length
+class AppDelegate: UIResponder, UIApplicationDelegate, CoronaWarnAppDelegate, RequiresAppDependencies, ENAExposureManagerObserver, CoordinatorDelegate, UNUserNotificationCenterDelegate, ExposureStateUpdating, ENStateHandlerUpdating {
```

Comment:
That's a great example why extensions for protocol implementations might be useful. Separate classes/structs for stuff like ENA observation or Notifications would be even nicer to reduce the size of this AppDelegate.